### PR TITLE
Add frequency and time masking and noise augmentation to AST

### PR DESCRIPTION
**Pull request by adrianfagerland**
# What does this PR do?

Adds frequency and time masking, as well as random noise data augmentations to AST feature extractor. This is very useful for training as AST has a tendency to overfit on practical data, at least in my own personal experience. This is also what was used in the original paper and [code](https://github.com/YuanGongND/ast/blob/master/src/dataloader.py#L187).

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [X] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request) Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)?
- [x] Did you make sure to update the documentation with your changes?
- [X] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed.

@ylacombe, @eustlb

This is my first time contributing to `transformers`, so please let me know if I need to make any changes :)

### CODE REVIEW

**Comment by ylacombe** [comment]
Hey @adrianfagerland, thanks for opening the PR, let me know when it's ready for review!

**Comment by adrianfagerland** [comment]
Ready for review now @ylacombe

**Comment by ylacombe** [comment]
Hey @adrianfagerland, sorry for the delay, let me know if you still want a review!

**Comment by adrianfagerland** [comment]
Sorry @ylacombe, I'm not able to reopen this PR because I deleted the forked repository so I had to make a new one, PR #34072.
